core/src/test/java/feign/UtilTest.java

```diff
@@ -122,6 +128,114 @@ public void unboundWildcardIsObject() throws Exception {
     assertEquals(Object.class, last);
   }
 
+  @Test
+  public void checkArgumentInputFalseNotNullNullOutputIllegalArgumentException() {
+    // Arrange
+    final boolean expression = false;
+    final String errorMessageTemplate = "";
+    final Object[] errorMessageArgs = null;
+    // Act
+    thrown.expect(IllegalArgumentException.class);
+    Util.checkArgument(expression, errorMessageTemplate, errorMessageArgs);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @Test
+  public void checkNotNullInputNullNotNullNullOutputNullPointerException() {
```

Comment:
For future reference, these would make a lot more sense if there was more separation of terms. All of this "NullNotNullNull" stuff makes zero sense to new eyes on the project, but "checkNotNull_referenceNull_messageNotNull_messageArgsNull_nullPointerExceptionThrown" is a lot easier to read and understand the coder's intentions without looking at the code in the test (then when looking at the code, you can more easily tell if it really tests what was intended).

Reply:
@rage-shadowman thanks for your feedback. Much appreciated.